tests/index-test.ts

```diff
@@ -300,6 +300,43 @@ describe('TSDI', () => {
       tsdi.get(Dependent);
     });
 
+    it('should call the initializer on an external after async dependencies w/o initializers are initialized', done => {
+      tsdi.enableComponentScanner();
+
+      @component
+      class DeepNestedAsyncDependency {
+        public value?: number;
+
+        @initialize
+        protected async init(): Promise<void> {
+          return new Promise<void>(resolve => {
+            setTimeout(() => {
+              this.value = 10;
+              resolve();
+            }, 10);
+          });
+        }
+      }
+
+      @component
+      class SyncDependency {
+        @inject public dependency!: DeepNestedAsyncDependency;
+      }
+
+      @external
+      class Dependent {
+        @inject public dependency!: SyncDependency;
+
+        @initialize
+        protected init(): void {
+          assert.equal(this.dependency.dependency.value, 10);
+          done();
+        }
+      }
+
+      new Dependent();
```

Comment:
@Prior99 This case is tricky. This is caused by the case that you get an instance from the constructor, but its not ready to be used. Also I cannot throw here, because this `new Dependent()` is usually expected and called by some external library (e.g. react).
So I'm unsure how to handle this. What do you think?

Reply:
Constructing it and not having it ready is always the case when using tsdi, the important thing is that the `@initialize` hook is first called, when the async dependcies are ready. Otherwise there is no way of ever being sure when to be able to use them.

Reply:
Yeah sure, but usually, when using a constructor, the returners instance is ready. This is the case with TSDI and synchronous dependencies.
In our case this can lead to barley understandable issues.

Reply:
I would still expect the constoructor to return directly, all I would need is the `@initialize` hook to be called later, when all dependencies are ready.

Reply:
Will be implemented in https://github.com/KnisterPeter/tsdi/pull/1085